client/components/button-group/index.jsx

```diff
@@ -9,7 +9,11 @@ const ButtonGroup = ( { busy = false, children, className = '', primary = false
 		'is-primary': primary,
 	} );
 
-	return <span className={ buttonGroupClasses }>{ children }</span>;
+	return (
+		<span className={ buttonGroupClasses } data-testid="button-group-id">
```

Comment:
I'd ideally suggest against introducing `data-testid` in components without a very good reason. It's also discouraged by `@testing-library/react` too, [see docs](https://testing-library.com/docs/queries/bytestid/):

> In the spirit of [the guiding principles](https://testing-library.com/docs/guiding-principles), it is recommended to use this only after the other queries don't work for your use case. Using data-testid attributes do not resemble how your software is used and should be avoided if possible. That said, they are way better than querying based on DOM structure or styling css class names. Learn more about data-testids from the blog post ["Making your UI tests resilient to change"](https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change)

That being said, have you considered taking a look at the WCAG (Web Content Accessibility Guidelines) recommendations for that purpose? [ARIA17](https://www.w3.org/TR/WCAG20-TECHS/ARIA17.html) solves that perfectly with a `role="group"`, and this is widely adopted throughout various component libraries ([Bootstrap](https://getbootstrap.com/docs/4.0/components/button-group/), [CoreUI](https://coreui.io/react/docs/components/button-group/)). Last but not least, `@testing-library/react` has a `*byRole` query family ([see docs](https://testing-library.com/docs/queries/byrole)) so that approach should work just fine for our testing purposes, while still contributing to improving the accessibility of the component.

Reply:
Some mind-bending stuff here. A couple of points:

1. in this doc https://www.w3.org/TR/WCAG20-TECHS/ARIA17.html#ARIA17-examples , it clearly says `For a group of radio buttons, one could also use role="radiogroup" instead of role="group"`
2. So I am creating a role called `buttongroup` but that is causing eslint to throw ![](https://d.pr/i/l69wmR+)
3. So I set the role to `button`, which is allowing me to use `getByRole` and  `getAllByRole` for all the 3 tests which seem to pass.

Reply:
Well, those aren't radio buttons, but are rather just plain buttons, so it's not a good example to look at IMHO. `role="group"` makes the most sense in my opinion. Also, see this document for more information and examples: https://www.digitala11y.com/group-role/ - you'll see an example that precisely displays a button that contains 2 buttons. On the other hand, having `role="button"` for the buttons makes a lot of sense, but we can't guarantee that the role of buttons in there will be "button", there could always be an additional component with a different role.